src/libvirtApi/domain.js

```diff
@@ -390,31 +374,22 @@ export function domainCreateFilesystem({ connectionName, objPath, vmName, source
     );
 }
 
-export function domainDelete({
+export async function domainDelete({
     connectionName,
     id: objPath,
     live,
 }) {
-    function destroy() {
-        return call(connectionName, objPath, 'org.libvirt.Domain', 'Destroy', [0], { timeout, type: 'u' });
-    }
+    if (live)
+        await call(connectionName, objPath, 'org.libvirt.Domain', 'Destroy', [0], { timeout, type: 'u' });
 
-    function undefine() {
+    try {
         const flags = Enum.VIR_DOMAIN_UNDEFINE_MANAGED_SAVE | Enum.VIR_DOMAIN_UNDEFINE_SNAPSHOTS_METADATA | Enum.VIR_DOMAIN_UNDEFINE_NVRAM;
 
-        return call(connectionName, objPath, 'org.libvirt.Domain', 'Undefine', [flags], { timeout, type: 'u' });
-    }
-
-    if (live) {
-        return destroy()
-                .then(undefine);
-    } else {
-        return undefine()
-                .catch(ex => {
-                    // Transient domains get undefined after shut off
-                    if (!ex.message.includes("Domain not found"))
-                        return Promise.reject(ex);
-                });
+        await call(connectionName, objPath, 'org.libvirt.Domain', 'Undefine', [flags], { timeout, type: 'u' });
+    } catch (ex) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1397-20240122-143818-61fe453f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)